Branch-SDK/src/io/branch/referral/BranchShortLinkBuilder.java

```diff
@@ -0,0 +1,152 @@
+ * Class for creating builder for getting a short url with Branch. This builder provide an easy and flexible way to configure and create
+ * a short url Synchronously or asynchronously.
+ * </p>
+ */
+public class BranchShortLinkBuilder extends BranchUrlBuilder {
+    public BranchShortLinkBuilder(Context context) {
+        super(context);
+    }
+
+    /**
+     * <p>Adds a tag to the iterable collection of name associated with a deep link.</p>
+     *
+     * @param tag {@link String} tags associated with a deep link.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder addTag(String tag) {
+        super.addTag(tag);
+        return this;
+    }
+
+    /**
+     * <p>Adds the the given key value pair to the parameters associated with this link.</p>
+     *
+     * @param key   A {@link String} with value of key for the parameter
+     * @param value A {@link String} with value of value for the parameter
+     * @returnA {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder addParameters(String key, String value) {
+        super.addParameters(key, value);
+        return this;
+    }
+
+    /**
+     * <p> Sets the alias for this link. </p>
+     *
+     * @param alias Link 'alias' can be used to label the endpoint on the link.
+     *              <p>
+     *              For example:
+     *              http://bnc.lt/AUSTIN28.
+     *              Should not exceed 128 characters
+     *              </p>
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setAlias(String alias) {
+        this.alias_ = alias;
+        return this;
+    }
+
+    /**
+     * <p> Sets the channel for this link. </p>
+     *
+     * @param channel A {@link String} denoting the channel that the link belongs to. Should not
+     *                exceed 128 characters.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setChannel(String channel) {
+        this.channel_ = channel;
+        return this;
+    }
+
+    /**
+     * <p> Sets the amount of time that Branch allows a click to remain outstanding.</p>
+     *
+     * @param duration A {@link Integer} value specifying the time that Branch allows a click to
+     *                 remain outstanding and be eligible to be matched with a new app session.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setDuration(int duration) {
+        this.duration_ = duration;
+        return this;
+    }
+
+    /**
+     * <p> Set a name that identifies the feature that the link makes use of.</p>
+     *
+     * @param feature A {@link String} value identifying the feature that the link makes use of.
+     *                Should not exceed 128 characters.
+     * @return A {@link BranchShortLinkBuilder} instance.
+     */
+    public BranchShortLinkBuilder setFeature(String feature) {
```

Comment:
For Referral url and Content url user has to `Channel` `Params` and `Tags` are the only optional item that can be set by dev